src/plugins/dashboard/public/application/state/dashboard_state_slice.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { Filter, Query } from '../../services/data';
+import { ViewMode } from '../../services/embeddable';
+import { DashboardOptions, DashboardPanelMap, DashboardState } from '../../types';
+
+export const dashboardStateSlice = createSlice({
+  name: 'counter',
+  initialState: {} as DashboardState,
+  reducers: {
+    setDashboardState: (state, action: PayloadAction<DashboardState>) => {
+      return action.payload;
+    },
+    updateState: (state, action: PayloadAction<Partial<DashboardState>>) => {
+      state = { ...state, ...action.payload };
```

Comment:
in those reducers, state is typed as `WritableDraft<DashboardState>`. I believe it's accomplished with some typescript magic on the side of Redux Toolkit.

Reply:
It's not really useful to have a default / initial state on the dashboard state, because it will get replaced immediately.

The true initial state is created async from the dashboard saved object, and I just consider the redux state totally blank before the saved object is loaded. I am not a big fan of the cast either, but it's required for redux toolkit to determine the type of the state.

Reply:
> in those reducers, state is typed as WritableDraft<DashboardState>

That what I was wondering because when I tested, for me it seemed that the test is untyped. But maybe it was because my typescript server was stuck during the branch switch 🤷‍♂️ 
I'd double-check that one